### PR TITLE
Social: Fix Bluesky profile URL

### PR DESCRIPTION
**Pull request by gmjuhasz**
Fixes https://github.com/Automattic/jetpack-reach/issues/653

## Proposed changes:

* Added the URL logic for Bluesky

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack-reach/issues/653
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Go to the connection management on the Admin page.
* You should see the external link on the profile name.
* Click on it, it should open the profile

CleanShot 2024-10-21 at 15 42 24 png

### CODE REVIEW

**Review by manzoorwanijk — CHANGES_REQUESTED** [review]
Our messy schema is cursed. It results in so many such bugs.

I wish we were consistent about when working on schemas.

We need to add another case here for missing name

image

**Comment by gmjuhasz** [comment]
@manzoorwanijk, should we fix that in it's own change, as this task is for fixing the profile URLs in the publicize package?

**Review by manzoorwanijk — APPROVED** [review]
Yeah, let us fix that schema in another change. Thanks

This works fine.

File: projects/packages/publicize/src/class-publicize-base.php [hunk 1]
```diff
@@ -517,6 +517,10 @@ public function get_profile_link( $service_name, $connection ) {
 			return 'https://twitter.com/' . substr( $cmeta['external_display'], 1 ); // Has a leading '@'.
 		}
 
+		if ( 'bluesky' === $service_name ) {
+			return 'https://bsky.app/profile/' . $cmeta['external_id'];
```
Comment (manzoorwanijk) [inline projects/packages/publicize/src/class-publicize-base.php:521]: Can we use `external_name` here instead?

Comment (gmjuhasz) [reply projects/packages/publicize/src/class-publicize-base.php:521]: Wouldn't that be the profiel name if you set up a display name? If so then that would not work

Comment (manzoorwanijk) [reply projects/packages/publicize/src/class-publicize-base.php:521]: No, `external_name` for now is always the handle.

Comment (gmjuhasz) [reply projects/packages/publicize/src/class-publicize-base.php:521]: I see! Do you think we should change to that or keep the Id?

Comment (manzoorwanijk) [reply projects/packages/publicize/src/class-publicize-base.php:521]: I think both are fine. Let us live with ID for now.

**Pull request merged**